### ISSUE TITLE
attribute-hyphenation should ignore attributes with the .prop modifier

### ISSUE THREAD

**Issue by danielgig** [vuejs/eslint-plugin-vue#685]
**Tell us about your environment**	
* **ESLint version:**	5.9.0
* **eslint-plugin-vue version:**	5.0.0-beta.5
* **Node version:**	 10.10.0

**Please show your full configuration:**

```js
module.exports = {
    extends: 'plugin:vue/recommended',
    parser: 'vue-eslint-parser',
    parserOptions: {
        parser: 'typescript-eslint-parser',
    },
    env: {
        browser: true,
    },
    rules: {
        'vue/attribute-hyphenation': [
            'warn',
            'always',
            {
                ignore: ['.prop']
            }
        ]
};
```

**What did you do?**	

```vue
<foo :barBaz.prop="val" />
```

**What did you expect to happen?**
I expect the linter to realise that since I'm using the `.prop` attribute it should respect the case of the attribute and not enforce kebab-case, because attributes with `.prop` are case sensitive and cannot be converted to kebab-case safely.

**What actually happened?**

`[eslint] Attribute ':barBaz.prop' must be hyphenated. [vue/attribute-hyphenation]`

**Comment by danielgig** [comment]
Thanks for your response. It turns out that the issue was caused by something else in the system, and vue properly converts kebab-case to camelCase even with `.prop`. You can close this issue as eslint-plugin-vue works fine. Sorry for the annoyance!

**Comment by michalsnik** [comment]
No worries :)
